src/EmbedIO/Files/FileModuleExtensions.cs

```diff
@@ -64,6 +64,27 @@ public static TModule WithContentCaching<TModule>(this TModule @this)
             return @this;
         }
 
+        /// <summary>
+        /// Enables caching of file contents and directory listings on a module.
+        /// </summary>
+        /// <typeparam name="TModule">The type of the module on which this method is called.</typeparam>
+        /// <param name="this">The module on which this method is called.</param>
+        /// <param name="maxFileSizeKb"><see langword="true"/> sets the maximum size of a single cached file in kilobytes;
+        /// <param name="maxSizeKb"><see langword="true"/> sets the maximum total size of cached data in kilobytes;
+        /// <returns><paramref name="this"/> with its <see cref="FileModule.ContentCaching">ContentCaching</see> property
+        /// set to <see langword="true"/>.</returns>
+        /// <exception cref="NullReferenceException"><paramref name="this"/> is <see langword="null"/>.</exception>
+        /// <exception cref="InvalidOperationException">The configuration of <paramref name="this"/> is locked.</exception>
+        /// <seealso cref="FileModule.ContentCaching"/>
+        public static TModule WithContentCaching<TModule>(this TModule @this, int maxFileSizeKb, int maxSizeKb)
+            where TModule : FileModule
+        {
+            @this.ContentCaching = true;
+            @this.MaxFileSizeKb = maxFileSizeKb;
+            @this.MaxSizeKb = maxSizeKb;
```

Comment:
Should be `@this.Cache.MaxSizeKb = maxSizeKb;`

Anyway, be aware that the cache is shared between all `FileModule`s by default.

Although each module gets its own section of the cache, the two properties you set take effect on the _whole_ cache. This fact would better be mentioned in XML docs too.